cv32e40s/tests/programs/custom/debug_test2/debug_test2.c

```diff
@@ -0,0 +1,3814 @@
+//
+// Copyright 2022 Silicon Labs, Inc.
+//
+// Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://solderpad.org/licenses/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+//
+// Author: Henrik Fegran
+//
+// Debug directed tests
+//
+/////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <string.h>
+#include <assert.h>
+#include "corev_uvmt.h"
+
+// MUST be 31 or less (bit position-1 in result array determines test pass/fail
+// status, thus we are limited to 31 tests with this construct.
+#define NUM_TESTS 19
+#define START_TEST_NUM 1
+// Abort test at first self-check fail, useful for debugging.
+#define ABORT_ON_ERROR_IMMEDIATE 0
```

Comment:
Possibly yes - but it has also proven beneficial at times during development, so I don't see the harm in leaving it in place, even though it might lead to different code behavior when enabled.

Reply:
Ok (y)